### PR TITLE
Website: Fix missing headings in ToC

### PR DESCRIPTION
**Pull request by Pandapip1**

### CODE REVIEW

File: _layouts/eip.html [hunk 1]
```diff
@@ -136,3 +136,6 @@ <h2>Citation</h2>
     "copyrightYear": "{{ page.created | date: "%Y" }}"
   }
 </script>
+<script type="application/javascript">
+  document.getElementsByClassName('footnotes')[0].insertAdjacentHTML('beforebegin', '<h2 id="footnotes">Footnotes</h2>');
```
Comment (SamWilsn) [inline _layouts/eip.html:140]: Won't this error if there are no footnotes?

Comment (Pandapip1) [reply _layouts/eip.html:140]: You're right. Let me fix that.

File: _layouts/eip.html [hunk 2]
```diff
@@ -136,3 +136,6 @@ <h2>Citation</h2>
     "copyrightYear": "{{ page.created | date: "%Y" }}"
   }
 </script>
+<script type="application/javascript">
+  [...document.getElementsByClassName('footnotes')].pop()?.insertAdjacentHTML('beforebegin', '<h2 id="footnotes">Footnotes</h2>');
```
Comment (SamWilsn) [inline _layouts/eip.html:140]: Would this work?

```suggestion
  document.getElementsByClassName('footnotes')[0]?.insertAdjacentHTML('beforebegin', '<h2 id="footnotes">Footnotes</h2>');
```

Comment (Pandapip1) [reply _layouts/eip.html:140]: Yes.

File: _includes/toc.html [hunk 3]
```diff
@@ -84,6 +84,9 @@
 {{ space }}{{ listModifier }} {{ listItemClass }} [{{ heading_body | replace: "|", "\|" }}]({% if include.baseurl %}{{ include.baseurl }}{% endif %}#{{ html_id }}){% if include.anchor_class %}{:.{{ include.anchor_class }}}{% endif %}{% endcapture %}
 {% endfor %}
 
+    - {{ listItemClass }} [Footnotes]({% if include.baseurl %}{{ include.baseurl }}{% endif %}#footnotes){% if include.anchor_class %}{:.{{ include.anchor_class }}}{% endif %}
```
Comment (SamWilsn) [inline _includes/toc.html:87]: This'll show a ToC entry even if the heading doesn't exist, right?

Comment (Pandapip1) [reply _includes/toc.html:87]: Yes. I see no good way to fix this, though.

Comment (SamWilsn) [reply _includes/toc.html:87]: Could you wrap it in a `<div style="display:none;">` and unhide it with the JavaScript that inserts the header?

Comment (Pandapip1) [reply _includes/toc.html:87]: I want to avoid JS at all costs.

Comment (Pandapip1) [reply _includes/toc.html:87]: It's probably worth me making a pre-processing step at some point. This seems like that time.

**Comment by Pandapip1** [comment]
Superseded by #6518

**Pull request closed without merging**